src/systems/behavior/dequeue.rs

```diff
@@ -0,0 +1,136 @@
+use std::fmt::Debug;
+
+use crate::components::{Idle, Queue};
+use bevy::prelude::{Commands, Component, Entity, Query, With};
+
+fn match_first<TBehavior: Copy, TComponent: TryFrom<TBehavior>>(
+	queue: &Queue<TBehavior>,
+) -> Option<TComponent> {
+	queue.0.get(0).and_then(|b| TComponent::try_from(*b).ok())
+}
+
+#[allow(clippy::type_complexity)]
```

Comment:
Really needed?

Reply:
On this one yes (otherwise I'd need to use some type indirections, that make it harder to understand imho), but I could remove the other occurrences.